### ISSUE TITLE
Deep Linking to location does not ask for location permission

### ISSUE THREAD

**Issue by sarahsporck** [digitalfabrik/integreat-app#2500]
### Describe the Bug

When deep linking to location, the user is not asked for location permission. This is also logged by the map library.

### Steps to Reproduce

1. Deep link to the map (http://localhost:9000/testumgebung/de/test-von-links-zu-orten)
2. Click on one of the links
3. See log message. Also you are not asked for location permission.

### Expected Behavior

Location permission is requested. Alternatively we could set permission to denied, when accessing map via deep link.

### Actual Behavior

Permission is not requested. Error is logged in the cmd.

### Additional Info
Check if this also happens for ios

**Comment by LeandraH** [comment]
iOS does request the location permission, so this only applies for Android.

**Comment by steffenkleinle** [comment]
Sadly I can't reproduce this, for me everything is working fine. @LeandraH could you reproduce this issue on android? Not sure what I am looking for here.

**Comment by LeandraH** [comment]
I could reproduce it on an emulated Pixel 7 API 34. I had location permissions for the app disabled in the system settings, then followed a location link in Testumgebung, and was immediately shown the map without any request for permissions. And I got the following error in the console: `locationManager Error:  [Error: uid 10191 does not have any of [android.permission.ACCESS_FINE_LOCATION, android.permission.ACCESS_COARSE_LOCATION].]`
[Screen_recording_no_permission.webm](https://github.com/digitalfabrik/integreat-app/assets/18513943/243c1ad2-2a77-4da1-8a9e-9188554f748c)

**Comment by steffenkleinle** [comment]
> I could reproduce it on an emulated Pixel 7 API 34. I had location permissions for the app disabled in the system settings, then followed a location link in Testumgebung, and was immediately shown the map without any request for permissions. And I got the following error in the console: `locationManager Error: [Error: uid 10191 does not have any of [android.permission.ACCESS_FINE_LOCATION, android.permission.ACCESS_COARSE_LOCATION].]` [Screen_recording_no_permission.webm](https://github.com/digitalfabrik/integreat-app/assets/18513943/243c1ad2-2a77-4da1-8a9e-9188554f748c)

Hmm, isn't this exactly the same behavior as if you open the map manually instead of via a deep link?

**Comment by LeandraH** [comment]
Good point, it is the same behaviour, sorry 😅  I'm not sure that asking every time would be a good user experience, I as a user would probably get pretty annoyed with it. And I think in iOS you natively can't ask for permission a second time if it was denied the first time, so we would have to build a workaround for that. I'd say that's not the best use of our time?

I guess the ticket might just be about fixing that warning in the console then, which really isn't a very high priority, I'd say. What do you think?

**Comment by steffenkleinle** [comment]
> Good point, it is the same behaviour, sorry 😅 I'm not sure that asking every time would be a good user experience, I as a user would probably get pretty annoyed with it. And I think in iOS you natively can't ask for permission a second time if it was denied the first time, so we would have to build a workaround for that. I'd say that's not the best use of our time?

Sorry, bad phrasing. What I meant: The map route should initially ask for the permission if not yet granted or disabled. If already granted or disabled, the current behavior should be kept. I wouldn't try to workaround and I also don't think its possible. However, not sure if that is even worth it though as clicking on the GeolocationIcon is doing the same.

> I guess the ticket might just be about fixing that warning in the console then, which really isn't a very high priority, I'd say. What do you think?

I think we shouldn't do even that since basically that is not a bug. Its just logging the current state, right?

So basically I think there is nothing to do here.

One other thing I noticed though: If the permission is denied, clicking on the GeolocationIcon does nothing. I think we should open the settings in that case. I will create an issue for that.

**Comment by steffenkleinle** [comment]
@sarahsporck what are your thoughts on this? Do you have a different opinion here?

No need to answer 💚

**Comment by ztefanie** [comment]
Current behaviour is the wanted behavior.
